### PR TITLE
[Feature] Adds PHPStan to the SDLC process

### PR DESCRIPTION
**Pull request by ahpaleus**
# Description

This Pull Request adds the [PHPStan](https://phpstan.org/) to the SDLC (as a GitHub Action) of the Snipe-IT project. PHPStan catches whole classes of bugs. The current configuration is to perform analysis on the master branch upon a push. Even though you use Codacy Security Scan - it uses only `PHP Mess Detector` and `PHP_CodeSniffer` for PHP, and we find it essential to implement other tools in your SDLC process.

- The configuration is in the phpstan.neon file
- Uses the [Larastan extension](https://github.com/nunomaduro/larastan) to cover Laravel's functions.
- Uses the [mcrypt](https://raw.githubusercontent.com/JetBrains/phpstorm-stubs/master/mcrypt/mcrypt.php) stub to cover deprecated functions
- Due to the verbosity, the results are shown under the Execute PHPStan in the specific GitHub Action.

Also, this Pull Requests contains some fixes for bugs identified by the PHPStan.

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

It was tested on the Snipe-IT's fork (on the master branch).

# Checklist:

- [X] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [X] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Console/Commands/LdapSync.php [hunk 1]
```diff
@@ -253,8 +253,8 @@ public function handle()
 
                 if ($item['ldap_location_override'] == true) {
                     $user->location_id = $item['location_id'];
-                } elseif ((isset($location)) && (! empty($location))) {
```
Comment (snipe) [inline app/Console/Commands/LdapSync.php]: Should these changes be in here? They don't seem related to code quality.

**Comment by snipe** [comment]
Please separate out these into two different PRs so they are easier to review. Separation of concerns and all that.

We already use PHPcs and phpmd locally and in Codacy (which gets run against every PR) as part of our existing SDLC, so I'm not exactly sure what benefit there is to this. 

We are waiting to run automated fixes until develop and master are merged (basically when v6 is out), so that we don't create a million conflicts that are all based around line breaks and indents.

File: app/Console/Commands/LdapSync.php (hunk 1 above)
Comment (ahpaleus) [reply app/Console/Commands/LdapSync.php]: Fixed

**Comment by snipe** [comment]
This PR conflicts with your previous PR. Please resolve the conflicts and we'll take another look.
